src/plane.rs

```diff
@@ -38,9 +38,105 @@ pub struct PlaneOffset {
   pub y: isize
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct PlaneData<T: Pixel> {
+  ptr: std::ptr::NonNull<T>,
+  _marker: PhantomData<T>,
+  layout: Layout,
+  len: usize,
+}
+
+unsafe impl<T: Pixel + Send> Send for PlaneData<T> { }
+unsafe impl<T: Pixel + Sync> Sync for PlaneData<T> { }
+
+impl<T: Pixel> Clone for PlaneData<T> {
+  fn clone(&self) -> Self {
+    let mut pd = unsafe { Self::new_uninitialized(self.len) };
+
+    pd.copy_from_slice(self);
+
+    pd
+  }
+}
+
+impl<T: Pixel> std::ops::Deref for PlaneData<T> {
+  type Target = [T];
+
+  fn deref(&self) -> &[T] {
+    unsafe {
+      let p = self.ptr.as_ptr();
+
+      std::slice::from_raw_parts(p, self.len)
+    }
+  }
+}
+
+impl<T: Pixel> std::ops::DerefMut for PlaneData<T> {
+  fn deref_mut(&mut self) -> &mut [T] {
+    unsafe {
+      let p = self.ptr.as_ptr();
+
+      std::slice::from_raw_parts_mut(p, self.len)
+    }
+  }
+}
+
+impl<T: Pixel> std::ops::Drop for PlaneData<T> {
+  fn drop(&mut self) {
+    unsafe {
+      dealloc(self.ptr.as_ptr() as *mut u8, self.layout);
+    }
+  }
+}
+
+impl<T: Pixel> PlaneData<T> {
+  /// Data alignment in bytes.
+  #[cfg(windows)]
+  const DATA_ALIGNMENT_LOG2: usize = 4;
+  #[cfg(not(windows))]
+  const DATA_ALIGNMENT_LOG2: usize = 5;
```

Comment:
Why is it different for Windows?

Reply:
This is rolled over from the existing aligned allocation. That gave SEGFAULT on win64 for `DATA_ALIGNMENT_LOG2=5`, presumably because of an illegal `dealloc` with unmatching alignment. We should be able to drop it safely now that we have our own container.